packages/salesforcedx-vscode-automation-tests/test/specs/createOASDoc.e2e.ts

```diff
@@ -0,0 +1,812 @@
+    } else {
+      await pause(Duration.seconds(5));
+    }
+    try {
+      await executeQuickPick('SFDX: Create OpenAPI Document from This Class (Beta)');
+      const failureNotificationWasFound = await notificationIsPresentWithTimeout(
+        /Failed to create OpenAPI Document: The Apex Class IneligibleApexClass is not valid for OpenAPI document generation\./,
+        Duration.TEN_MINUTES
+      );
+      expect(failureNotificationWasFound).to.equal(true);
+    } catch (error) {
+      await pause(Duration.minutes(1));
+      await executeQuickPick('SFDX: Create OpenAPI Document from This Class (Beta)');
+      const failureNotificationWasFound = await notificationIsPresentWithTimeout(
+        /Failed to create OpenAPI Document: The Apex Class IneligibleApexClass is not valid for OpenAPI document generation\./,
+        Duration.TEN_MINUTES
+      );
+      expect(failureNotificationWasFound).to.equal(true);
+    }
+  });
+
+  describe('Composed mode', async () => {
+    step(
+      'Generate OAS doc from a valid Apex class using command palette - Composed mode, initial generation',
+      async () => {
+        log(
+          `${testSetup.testSuiteSuffixName} - Generate OAS doc from a valid Apex class using command palette - Composed mode, initial generation`
+        );
+        await executeQuickPick('View: Close All Editors');
+        await openFile(
+          path.join(testSetup.projectFolderPath!, 'force-app', 'main', 'default', 'classes', 'CaseManager.cls')
+        );
+        await pause(Duration.seconds(5));
+        prompt = await executeQuickPick('SFDX: Create OpenAPI Document from This Class (Beta)');
+        await prompt.confirm();
+
+        const successNotificationWasFound = await notificationIsPresentWithTimeout(
+          /OpenAPI Document created for class: CaseManager\./,
+          Duration.TEN_MINUTES
+        );
+        expect(successNotificationWasFound).to.equal(true);
+
+        // Verify the generated OAS doc is open in the Editor View
+        await executeQuickPick('View: Open Last Editor in Group');
+        const workbench = getWorkbench();
+        const editorView = workbench.getEditorView();
+        const activeTab = await editorView.getActiveTab();
+        const title = await activeTab?.getTitle();
+        expect(title).to.equal('CaseManager.externalServiceRegistration-meta.xml');
+      }
+    );
+
+    step('Check for warnings and errors in the Problems Tab', async () => {
+      log(`${testSetup.testSuiteSuffixName} - Check for warnings and errors in the Problems Tab`);
+      await countProblemsInProblemsTab(0);
+    });
+
+    step('Fix the OAS doc to get rid of the problems in the Problems Tab', async () => {
+      // NOTE: The "fix" is actually replacing the OAS doc with the ideal solution
+      log(`${testSetup.testSuiteSuffixName} - Fix the OAS doc to get rid of the problems in the Problems Tab`);
+
+      const idealCaseManagerOASDoc = [
```

Comment:
these should probably be in a separate file.  you often see `__fixtures__` or `/samples` etc.

instead of writing giant strings in code